superset-frontend/src/components/Select/utils.ts

```diff
@@ -61,27 +60,25 @@ export function findValue<OptionType extends OptionTypeBase>(
   return (Array.isArray(value) ? value : [value]).map(find);
 }
 
-export function hasOption<VT extends string | number>(
-  value: VT,
-  options?: VT | VT[] | { value: VT } | { value: VT }[],
-) {
+export function getValue(option: string | number | { value: string | number }) {
+  return typeof option === 'object' ? option.value : option;
+}
+
+type LabeledValue<V> = { label?: ReactNode; value?: V };
+
+export function hasOption<V>(
+  value: V,
+  options?: V | LabeledValue<V> | (V | LabeledValue<V>)[],
+  checkLabel = false,
+): boolean {
   const optionsArray = ensureIsArray(options);
   return (
-    optionsArray.find(x =>
-      typeof x === 'object' ? x.value === value : x === value,
+    optionsArray.find(
+      x =>
+        x === value ||
+        (typeof x === 'object' &&
+          (('value' in x && x.value === value) ||
+            (checkLabel && 'label' in x && x.label === value))),
```

Comment:
Just a nit but I think it's better for readability 

```suggestion
    optionsArray.find(
      option =>
        option === value ||
        (typeof option === 'object' &&
          (('value' in option && option.value === value) ||
            (checkLabel && 'label' in option && option.label === value))),
```

Reply:
See my comment in https://github.com/apache/superset/pull/18799#discussion_r818903790